examples/specialty_plots/ishikawa.py

```diff
@@ -0,0 +1,229 @@
+                                    xytext=(x_start, y_start),
+                                    arrowprops=dict(arrowstyle="->",
+                                                    facecolor=spine_color),
+                                    bbox=props, weight='bold')
+            else:
+                props = dict(boxstyle='round', facecolor='lavender', alpha=1.0)
+                spine = ax.annotate(k, xy=(x_end, y_end), xytext=(x_start, y_start),
+                                    arrowprops=dict(arrowstyle="->",
+                                                    facecolor=spine_color),
+                                    bbox=props)
+            # Call recursion to draw subspines
+            drawspines(data=data[k], ax=ax, parentspine=spine,
+                       alpha_ps=alpha_ps, alpha_ss=alpha_ss,
+                       spine_color=spine_color, reclevel=reclevel + 1,
+                       max_recursion_level=max_recursion_level)
+            # no primary_spines_rel_xpos is needed to be passed on recursion
+            # next spine settings - same level
+            alpha *= -1
+            spine_count += 1
+            if spine_count % 2 == 0:
+                s = s + spacing
+    return None
+
+
+def ishikawaplot(data: dict, ax: plt.Axes, left_margin: float = 0.05,
+                 right_margin: float = 0.05, alpha_ps: float = 60.0,
+                 alpha_ss=0.0, primary_spines_rel_xpos: List[float] = [np.nan],
+                 pd_width: int = 0.1, spine_color: str = "black") -> None:
+    """
+
+    Parameters
+    ----------
+    data : dict
+        Plot data structure.
+    ax : matplotlib.pyplot.Axes
+        Axes in which to drow the plot.
+    left_margin : float, optional
+        Left spacing from frame border. The default is 0.05.
+    right_margin : float, optional
+        Right spacing from frame border. The default is 0.05.
+    alpha_ps : float, optional
+        First spine angle using during recursion. The default is 60.0.
+    alpha_ss : TYPE, optional
+        Secondary spine angle using during recursion. The default is 0.0.
+    primary_spines_rel_xpos : list, optional
+        Relative X-position of primary spines, a list of float is accepted.
+        The default is [np.nan].
+    pd_width : int, optional
+        Problem description box relative width. The default is 0.1.
+    spine_color : str, optional
+        Spine color. The default is "black".
+
+    Returns
+    -------
+    None
+
+    """
+    # format axis
+    ax.set_xlim(0, 1.0)
+    ax.set_ylim(0, 1.0)
+    ax.axis('off')
+
+    # draw main spine
+    main_spine = ax.axhline(y=0.5, xmin=left_margin,
+                            xmax=1 - right_margin - pd_width,
+                            color=spine_color)
+
+    # draw fish head
+    props = dict(boxstyle='round', facecolor='wheat', alpha=1.0)
+    # add problem tag
+    ax.text((1 - right_margin - pd_width), 0.5, str.upper(list(data.keys())[0]),
```

Comment:
I'm not sure picking the first key (and ignoring everything else) from the `data` is the best choice? Why not pass a name and causes separately instead?

Reply:
Please see comment about JSON formatting below